src/actions/deviceActions.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2015 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import {
+    Device,
+    DeviceSetup,
+    getAppFile,
+    jprogDeviceSetup,
+    logger,
+    prepareDevice,
+    sdfuDeviceSetup,
+} from 'pc-nrfconnect-shared';
+import { SerialPort } from 'serialport';
+
+import {
+    clearSerialPort,
+    setAvailableSerialPorts,
+    setSelectedSerialport,
+    setSerialPort,
+} from '../features/rssiSlice';
+import { TAction } from '../thunk';
+
+export const deviceSetup: DeviceSetup = {
+    deviceSetups: [
+        sdfuDeviceSetup([
+            {
+                key: 'pca10059',
+                application: getAppFile('fw/rssi-10059.hex'),
+                semver: 'rssi_cdc_acm 2.0.0+dfuMay-22-2018-10-43-22',
+                params: {},
+            },
+        ]),
+        jprogDeviceSetup([
+            {
+                key: 'nrf52_family',
+                fw: getAppFile('fw/rssi-10040.hex'),
+                fwVersion: 'rssi-fw-1.0.0',
+                fwIdAddress: 0x2000,
+            },
+        ]),
+    ],
+    needSerialport: true,
+};
```

Comment:
If this is here only because of the recoverHex, I'd prefer to have it in a separate file.